\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage[margin=1in]{geometry}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={AmSam\_corals\_2020},
            pdfauthor={Cheryl Logan},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{AmSam\_corals\_2020}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{Cheryl Logan}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{5/5/2020}


\begin{document}
\maketitle

\hypertarget{run-qc-on-raw-rnaseq-reads-in-unix}{%
\subsection{Run QC on raw RNAseq reads in
unix}\label{run-qc-on-raw-rnaseq-reads-in-unix}}

Similar to the FastX Toolobox, we will use a program called Trimmomatic
to both QC and pair raw reads. In the following example, I show how to
run Trimmomatic on a single end (SE) file and paired end (PE) files. Our
coral samples are 150 bp PE reads, so we will use the PE mode.

Single End (SE) Mode:

\begin{Shaded}
\begin{Highlighting}[]

\ExtensionTok{java}\NormalTok{ -jar /opt/Trimmomatic-0.35/trimmomatic-0.35.jar SE }\DataTypeTok{\textbackslash{} }\NormalTok{          # call program using SE option}
\ExtensionTok{-phred33} \DataTypeTok{\textbackslash{} }\NormalTok{                                                             # Illumina quality score}
\ExtensionTok{CoPtE1S_CKDL200153111-1B_H7WTWBBXX_L4_1.fq.gz}\NormalTok{ CoPtE1S_1_trimmo.fq.gz }\DataTypeTok{\textbackslash{} } \CommentTok{# input/output filenames}
\ExtensionTok{ILLUMINACLIP}\NormalTok{:/opt/Trimmomatic-0.35/adapters/TruSeq3-SE.fa:2:30:10 }\DataTypeTok{\textbackslash{} }\NormalTok{    # adapter removal}
\ExtensionTok{SLIDINGWINDOW}\NormalTok{:4:5 LEADING:5 TRAILING:5 MINLEN:25                        # poor quality base removal}
\CommentTok{# careful when copying code to command line (check for extra spaces)}
\end{Highlighting}
\end{Shaded}

Paired End (SE) Mode:

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{java}\NormalTok{ -jar /opt/Trimmomatic-0.35/trimmomatic-0.35.jar PE }\DataTypeTok{\textbackslash{} } \CommentTok{# call program using PE option}
\ExtensionTok{-phred33} \DataTypeTok{\textbackslash{} }\NormalTok{                                                # Illumina quality score}
\ExtensionTok{-threads}\NormalTok{ 4 }\DataTypeTok{\textbackslash{} }\NormalTok{                                              # Use 4 computing threads to speed up run}
\ExtensionTok{CoPtE1S_CKDL200153111-1B_H7WTWBBXX_L4_1.fq.gz}\NormalTok{ CoPtE1S_CKDL200153111-1B_H7WTWBBXX_L4_2.fq.gz  }\DataTypeTok{\textbackslash{} }\NormalTok{                                   # input R1 and R2 fileneames}
\ExtensionTok{CoPtE1S_1.trimmed.fq.gz}\NormalTok{ CoPtE1S_1un.trimmed.fq.gz }\DataTypeTok{\textbackslash{} }\NormalTok{  # output filenames for paired }\KeywordTok{&} \ExtensionTok{unpaired}\NormalTok{ R1}
\ExtensionTok{CoPtE1S_2.trimmed.fq.gz}\NormalTok{ CoPtE1S_2un.trimmed.fq.gz }\DataTypeTok{\textbackslash{} }\NormalTok{  # output filenames for paired }\KeywordTok{&} \ExtensionTok{unpaired}\NormalTok{ R2}
\ExtensionTok{ILLUMINACLIP}\NormalTok{:/opt/Trimmomatic-0.35/adapters/TruSeq3-PE.fa:2:30:10 }\DataTypeTok{\textbackslash{} }\NormalTok{  # adapter removal}
\ExtensionTok{SLIDINGWINDOW}\NormalTok{:4:5 LEADING:5 TRAILING:5 MINLEN:25           # poor quality base removal}
\end{Highlighting}
\end{Shaded}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  Trimmomatic does the following steps in order: Remove Illumina
  adapters provided in TruSeq3-SE.fa or TruSeq3-PE.fa note: Trimmomatic
  will look for seed matches (16 bases) allowing maximally 2 mismatches.
  These seeds will be extended and clipped if in the case of paired end
  reads a score of 30 is reached (about 50 bases), or in the case of
  single ended reads a score of 10 (about 17 bases)
\item
  Remove leading low quality or N bases (below quality 5)
\item
  Remove trailing low quality or N bases (below quality 5)
\item
  Scan the read with a 4 base wide sliding window, cutting when the
  average quality per base drops below 5
\item
  Drop reads which are less than 25 bases long after these steps
\end{enumerate}

Trimmomatic manual here:
\url{http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf}

\hypertarget{biomsci430-quick-qc-pipeline}{%
\section{BIO/MSCI430 Quick QC
Pipeline}\label{biomsci430-quick-qc-pipeline}}

\begin{itemize}
\tightlist
\item
  To quickly generate results for the class, I ran the bioinformatics
  pipeline on 1 sample per treatment (n=6 total)
\item
  This summer, Melissa and Silvia will repeat this analysis for all the
  samples from the class experiment
\item
  cd into each sample directory and modify code for each sample as
  follows (or create bash file for all and run on command line with
  nohup)
\item
  after trimming is complete, move output files to a separate QC folder
\item
  see how I modified the trimmomatic PE code above with each of the 6
  samples below
\end{itemize}

\hypertarget{copte1s}{%
\section{CoPtE1S}\label{copte1s}}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{java}\NormalTok{ -jar /opt/Trimmomatic-0.35/trimmomatic-0.35.jar PE -phred33 -threads 4 CoPtE1S_CKDL200153111-1B_H7WTWBBXX_L4_1.fq.gz CoPtE1S_CKDL200153111-1B_H7WTWBBXX_L4_2.fq.gz CoPtE1S_1.trimmed.fq.gz CoPtE1S_1un.trimmed.fq.gz CoPtE1S_2.trimmed.fq.gz CoPtE1S_2un.trimmed.fq.gz ILLUMINACLIP:/opt/Trimmomatic-0.35/adapters/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:4:5 LEADING:5 TRAILING:5 MINLEN:25}

\FunctionTok{mv}\NormalTok{ *.trim* /home/data/corals/QCdata}
\end{Highlighting}
\end{Shaded}

\hypertarget{copth1s}{%
\section{CoPtH1S}\label{copth1s}}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{java}\NormalTok{ -jar /opt/Trimmomatic-0.35/trimmomatic-0.35.jar PE -phred33 -threads 4 CoPtH1S_CKDL200153118-1B_H7WTWBBXX_L4_1.fq.gz CoPtH1S_CKDL200153118-1B_H7WTWBBXX_L4_2.fq.gz CoPtH1S_1.trimmed.fq.gz CoPtH1S_1un.trimmed.fq.gz CoPtH1S_2.trimmed.fq.gz CoPtH1S_2un.trimmed.fq.gz ILLUMINACLIP:/opt/Trimmomatic-0.35/adapters/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:4:5 LEADING:5 TRAILING:5 MINLEN:25}

\FunctionTok{mv}\NormalTok{ *.trim* /home/data/corals/QCdata}
\end{Highlighting}
\end{Shaded}

\hypertarget{falue1s}{%
\section{FaluE1S}\label{falue1s}}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{java}\NormalTok{ -jar /opt/Trimmomatic-0.35/trimmomatic-0.35.jar PE -phred33 -threads 4 FaluE1S_CKDL200153096-1B_H7WTWBBXX_L4_1.fq.gz FaluE1S_CKDL200153096-1B_H7WTWBBXX_L4_2.fq.gz FaluE1S_1.trimmed.fq.gz FaluE1S_1un.trimmed.fq.gz FaluE1S_2.trimmed.fq.gz FaluE1S_2un.trimmed.fq.gz ILLUMINACLIP:/opt/Trimmomatic-0.35/adapters/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:4:5 LEADING:5 TRAILING:5 MINLEN:25}

\FunctionTok{mv}\NormalTok{ *.trim* /home/data/corals/QCdata}
\end{Highlighting}
\end{Shaded}

\hypertarget{faluh1s}{%
\section{FaluH1S}\label{faluh1s}}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{java}\NormalTok{ -jar /opt/Trimmomatic-0.35/trimmomatic-0.35.jar PE -phred33 -threads 4 FaluH1S_CKDL200153097-1B_H7WTWBBXX_L4_1.fq.gz FaluH1S_CKDL200153097-1B_H7WTWBBXX_L4_2.fq.gz FaluH1S_1.trimmed.fq.gz FaluH1S_1un.trimmed.fq.gz FaluH1S_2.trimmed.fq.gz FaluH1S_2un.trimmed.fq.gz ILLUMINACLIP:/opt/Trimmomatic-0.35/adapters/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:4:5 LEADING:5 TRAILING:5 MINLEN:25}

\FunctionTok{mv}\NormalTok{ *.trim* /home/data/corals/QCdata}
\end{Highlighting}
\end{Shaded}

\hypertarget{ftelee3s}{%
\section{FteleE3S}\label{ftelee3s}}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{java}\NormalTok{ -jar /opt/Trimmomatic-0.35/trimmomatic-0.35.jar PE -phred33 -threads 4 FteleE3S_CKDL200153113-1B_H7WTWBBXX_L4_1.fq.gz FteleE3S_CKDL200153113-1B_H7WTWBBXX_L4_2.fq.gz FteleE3S_1.trimmed.fq.gz FteleE3S_1un.trimmed.fq.gz FteleE3S_2.trimmed.fq.gz FteleE3S_2un.trimmed.fq.gz ILLUMINACLIP:/opt/Trimmomatic-0.35/adapters/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:4:5 LEADING:5 TRAILING:5 MINLEN:25}

\FunctionTok{mv}\NormalTok{ *.trim* /home/data/corals/QCdata}
\end{Highlighting}
\end{Shaded}

\hypertarget{fteleh3s}{%
\section{FteleH3S}\label{fteleh3s}}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{java}\NormalTok{ -jar /opt/Trimmomatic-0.35/trimmomatic-0.35.jar PE -phred33 -threads 4 FteleH3S_CKDL200153119-1B_H7WTWBBXX_L4_1.fq.gz FteleH3S_CKDL200153119-1B_H7WTWBBXX_L4_2.fq.gz FteleH3S_1.trimmed.fq.gz FteleH3S_1un.trimmed.fq.gz FteleH3S_2.trimmed.fq.gz FteleH3S_2un.trimmed.fq.gz ILLUMINACLIP:/opt/Trimmomatic-0.35/adapters/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:4:5 LEADING:5 TRAILING:5 MINLEN:25}

\FunctionTok{mv}\NormalTok{ *.trim* /home/data/corals/QCdata}
\end{Highlighting}
\end{Shaded}

move QC\_data folder and assembly from treebeard to khaleesi server

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{scp}\NormalTok{ -r /home/data/corals/QC_data username@khaleesi.treebeard.csumb.edu:/data/corals/QCdata }\CommentTok{# use -r with scp to move folders}
\end{Highlighting}
\end{Shaded}

\hypertarget{mapping-counting-using-rsem}{%
\subsection{Mapping \& Counting using
RSEM}\label{mapping-counting-using-rsem}}

After QC, the next step is to map our reads to a de novo transcriptome
assembly and count how many time each transcript is expressed in each
sample. The Acropora hyacinthus de novo transcriptome assembly we used
is from Barshis et al.~2013 PNAS. We will use two programs, bowtie2 and
RSEM, to respectively map and count our reads to this assembly. (Ran all
remaining scripts on khaleesi server)

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nohup}\NormalTok{ perl /opt/trinityrnaseq/util/align_and_estimate_abundance.pl --transcripts 33496_Ahyacinthus_CoralContigs.fasta --seqType fq --samples_file samples_class.txt --est_method RSEM --aln_method bowtie2 --prep_reference --output_dir /data/corals/QCdata/mapping }\OperatorTok{>}\NormalTok{ RSEM_out }\OperatorTok{2>&1} \KeywordTok{&}
\CommentTok{# nohup allows for longer processes to continue running even when you are not logged into server}
\CommentTok{# what would normally be printed on the terminal is saved in 'RSEM_out'}
\CommentTok{# [1] 31382}
\CommentTok{# [1] 32754}
\end{Highlighting}
\end{Shaded}

\hypertarget{build-transcript-and-gene-expression-matrices}{%
\section{Build Transcript and Gene Expression
Matrices}\label{build-transcript-and-gene-expression-matrices}}

This step builds a counts matrix, providing the number of times each
gene was expressed in each sample.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nohup}\NormalTok{ perl /opt/trinityrnaseq/util/abundance_estimates_to_matrix.pl --est_method RSEM  --gene_trans_map none --name_sample_by_basedir CoPt_E_rep1/RSEM.genes.results CoPt_H_rep1/RSEM.genes.results Falu_E_rep1/RSEM.genes.results Falu_H_rep1/RSEM.genes.results Ftele_E_rep3/RSEM.genes.results Ftele_H_rep3/RSEM.genes.results }\OperatorTok{>}\NormalTok{ RSEMae_out }\OperatorTok{2>&1} \KeywordTok{&}
\end{Highlighting}
\end{Shaded}

\hypertarget{compare-replicates}{%
\section{Compare Replicates}\label{compare-replicates}}

note 1: we did not use a Trinity assembly, so isoform here actually
refers to gene note 2: for the quick class pipeline, we only have n=1
here, so these commands are only relevant when using replicates

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{perl}\NormalTok{ /opt/trinityrnaseq/Analysis/DifferentialExpression/PtR --matrix RSEM.isoform.counts.matrix --min_rowSums 10 -s samples_class.txt --log2 --CPM --sample_cor_matrix}

\FunctionTok{perl}\NormalTok{ /opt/trinityrnaseq/Analysis/DifferentialExpression/PtR --matrix RSEM.isoform.counts.matrix -s samples_class.txt --min_rowSums 10 --log2 --CPM --center_rows --prin_comp 3 }
\end{Highlighting}
\end{Shaded}

\hypertarget{differential-gene-expression-analysis-using-edger}{%
\subsection{Differential Gene Expression Analysis using
edgeR}\label{differential-gene-expression-analysis-using-edger}}

This step uses program called edgeR to identify which genes were
differentially expression among our treatment groups.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# with no replicates (For class only!! We would use ALL replicates for full analysis!)}
\FunctionTok{perl}\NormalTok{ /opt/trinityrnaseq/Analysis/DifferentialExpression/run_DE_analysis.pl --matrix RSEM.isoform.counts.matrix --method edgeR --dispersion 0.1}

\CommentTok{# cd into new edgeR folder, then run DGE}
\FunctionTok{perl}\NormalTok{ /opt/trinityrnaseq/Analysis/DifferentialExpression/analyze_diff_expr.pl --matrix /data/corals/QCdata/RSEM.isoform.TMM.EXPR.matrix -P 1e-3 -C 2}
\end{Highlighting}
\end{Shaded}

\hypertarget{resources}{%
\section{Resources}\label{resources}}

\hypertarget{more-on-trimmomatic}{%
\subsection{More on Trimmomatic}\label{more-on-trimmomatic}}

note: Trimmomatic will look for seed matches (16 bases) allowing
maximally 2 mismatches. These seeds will be extended and clipped if in
the case of paired end reads a score of 30 is reached (about 50 bases),
or in the case of single ended reads a score of 10 (about 17 bases) 2.
Remove leading low quality or N bases (below quality 5) 3. Remove
trailing low quality or N bases (below quality 5) 4. Scan the read with
a 4 base wide sliding window, cutting when the average quality per base
drops below 5 5. Drop reads which are less than 25 bases long after
these steps

Trimmomatic manual here:
\url{http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\setcounter{enumi}{1}
\tightlist
\item
  How to Run Trimmomatic on all fastq files in a directory Tips:
  \url{https://datacarpentry.org/wrangling-genomics/03-trimming/}
\end{enumerate}

We can use a for loop in bash to run trimmomatic on all the files ending
in .fastq and appending the original file name with `.trimmo.fq.gz'

\begin{Shaded}
\begin{Highlighting}[]
\KeywordTok{for} \ExtensionTok{infile}\NormalTok{ in *_1.fq.gz}
\KeywordTok{do}
   \VariableTok{base=$(}\FunctionTok{basename} \VariableTok{$\{infile\}}\NormalTok{ _1.fastq.gz}\VariableTok{)}
   \ExtensionTok{java}\NormalTok{ -jar /opt/Trimmomatic-0.35/trimmomatic-0.35.jar PE -phred33 -threads 4 \textbackslash{}}
                \VariableTok{$\{infile\}} \VariableTok{$\{base\}}\NormalTok{_2.fastq.gz \textbackslash{}}
                \VariableTok{$\{base\}}\NormalTok{_1.trim.fastq.gz }\VariableTok{$\{base\}}\NormalTok{_1un.trim.fastq.gz \textbackslash{}}
                \VariableTok{$\{base\}}\NormalTok{_2.trim.fastq.gz }\VariableTok{$\{base\}}\NormalTok{_2un.trim.fastq.gz \textbackslash{}}
\NormalTok{                ILLUMINACLIP:/opt/Trimmomatic-0.35/adapters/TruSeq3-PE.fa:2:30:10 SLIDINGWINDOW:4:5 LEADING:5 TRAILING:5 MINLEN:25}

\KeywordTok{done}
\end{Highlighting}
\end{Shaded}

\hypertarget{using-fastqc-to-inspect-reads-prepost-qc}{%
\subsection{Using FastQC to inspect reads pre/post
QC}\label{using-fastqc-to-inspect-reads-prepost-qc}}

Always inspect and compare your files pre- and post-QC. Students should
make a table describing the FastQC results on files pre/postQC. For more
info on this tool, go to their website:

\url{https://www.bioinformatics.babraham.ac.uk/projects/fastqc/}

\begin{Shaded}
\begin{Highlighting}[]
\ExtensionTok{fastqc}\NormalTok{ filename.fastq}
\ExtensionTok{fastqc}\NormalTok{ filename_trimmo.fastq}
\end{Highlighting}
\end{Shaded}

You can download the FastQC report files and view them on your computer.
Use the \texttt{scp} command in a terminal window to do this. Before you
run this command, navigate the the folder on your computer where you
want to download the files. Do not run this command on the server!

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{scp}\NormalTok{ otterID@treebeard.csumb.edu:/file/path/to/your.stuff/* .}
\end{Highlighting}
\end{Shaded}

\hypertarget{references}{%
\section{References}\label{references}}

Barshis, D. J., Ladner, J. T., Oliver, T. A., Seneca, F. O.,
Traylor-Knowles, N., \& Palumbi, S. R. (2013). Genomic basis for coral
resilience to climate change. Proceedings of the National Academy of
Sciences, 110(4), 1387-1392.

Bolger, A. M., Lohse, M., \& Usadel, B. (2014). Trimmomatic: a flexible
trimmer for Illumina sequence data. Bioinformatics, 30(15), 2114-2120.

Haas, B. J., Papanicolaou, A., Yassour, M., Grabherr, M., Blood, P. D.,
Bowden, J., \ldots{} \& MacManes, M. D. (2013). De novo transcript
sequence reconstruction from RNA-seq using the Trinity platform for
reference generation and analysis. Nature protocols, 8(8), 1494.

Li, B., \& Dewey, C. N. (2011). RSEM: accurate transcript quantification
from RNA-Seq data with or without a reference genome. BMC
bioinformatics, 12(1), 323.

MacManes, M. D. (2014). On the optimal trimming of high-throughput mRNA
sequence data. Frontiers in genetics, 5, 13.

Robinson, M. D., McCarthy, D. J., \& Smyth, G. K. (2010). edgeR: a
Bioconductor package for differential expression analysis of digital
gene expression data. Bioinformatics, 26(1), 139-140.

Note: there is no citation for FASTQC, so please refer to their website
in your reports
(\url{https://www.bioinformatics.babraham.ac.uk/projects/fastqc/})

Mapping, Counting, and DGE steps using Trinity:
\url{https://github.com/trinityrnaseq/trinityrnaseq/wiki/Trinity-Differential-Expression}


\end{document}
